tests/application/test_use_cases.py

```diff
@@ -45,10 +63,10 @@ def test_list_a_single_data_provider(app_context):
     name = "State X"
     add_data_provider(app_context, name)
 
-    providers = list_data_providers(app_context)
+    result = list_data_providers(app_context)
 
-    assert len(providers) == 1
-    assert providers[0].name == name
+    assert len(result) == 1
+    assert isinstance(result[0], DataProviderViewModel)
```

Comment:
Maybe there's coverage for this elsewhere, but the `assert providers[0].name == name` that was removed still seems valuable here. (same with the `test_list_multiple_data_providers` asserts)